File: SaliScoreDesktop/score/CsDefinition.cpp
#include "CsDefinition.h"

CsDefinition::CsDefinition(const QString nm, const QString descr) :
  mName(nm),
  mDescription(descr),
  mVisible(true)
  {

  }

void CsDefinition::jsonWrite(CsJsonWriter &js) const
  {
  js.jsonString( "Name", mName );
  js.jsonString( "Description", mDescription );
  js.jsonBool( "Visible", mVisible );
  }

void CsDefinition::jsonRead(CsJsonReader &js)
  {
  js.jsonString( "Name", mName );
  js.jsonString( "Description", mDescription );
  js.jsonBool( "Visible", mVisible );
  }

File: SaliScoreAndroid/app/src/main/java/salilab/saliscore/CsDefinition.java
package salilab.saliscore;

public class CsDefinition extends CsJsonWritable {
  public String  mName;
  public String  mDescription;
  public Boolean mVisible;

  CsDefinition() {
    mName = new String();
    mDescription = new String();
    mVisible = true;
  }

  CsDefinition( CsDefinition df ) {
    mName = df.mName;
    mDescription = df.mDescription;
    mVisible = df.mVisible.booleanValue();
  }

  CsDefinition( String nm, String descr ) {
    mName = nm;
    mDescription = descr;
    mVisible = true;
  }

  boolean visibleToggle() { return mVisible = !mVisible; }

  @Override
  void jsonWrite( SvJsonWriter js ) throws Exception {
    js.jsonString( "Name", mName );
    js.jsonString( "Description", mDescription );
    js.jsonBool( "Visible", mVisible );
  }

  @Override
  void jsonRead( SvJsonReader js ) throws Exception {
    js.jsonString( "Name", mName );
    js.jsonString( "Description", mDescription );
    js.jsonBool( "Visible", mVisible );
  }

  @Override
  CsJsonWritable copy() {
    return new CsDefinition( this );
  }
}
